src/openshift/component.ts

```diff
@@ -145,6 +144,67 @@ export class Component extends OpenShiftItem {
         Component.odo.executeInTerminal(Command.showLogAndFollow(component.getParent().getParent().getName(), component.getParent().getName(), component.getName()), component.contextPath.fsPath);
     }
 
+    private static async getLinkData(component: OpenShiftObject) {
+        const compData = await Component.odo.execute(Command.describeComponentJson(component.getParent().getParent().getName(), component.getParent().getName(), component.getName()), component.contextPath ? component.contextPath.fsPath : Platform.getUserHomePath());
+        return JSON.parse(compData.stdout);
+    }
+
+    static async unlink(context: OpenShiftObject) {
+        const unlinkActions = [
+            {
+                label: 'Component',
+                description: 'Unlink Component'
+            },
+            {
+                label: 'Service',
+                description: 'Unlink Service'
+            }
+        ];
+        const unlinkActionSelected = await window.showQuickPick(unlinkActions, {placeHolder: 'Select the option'});
+        if (!unlinkActionSelected) return null;
+        return unlinkActionSelected.label === 'Component' ? Component.unlinkComponent(context) : unlinkActionSelected.label === 'Service' ?  Component.unlinkService(context) : null;
+    }
+
+    static async unlinkComponent(context: OpenShiftObject) {
+        const linkCompName: Array<string> = [];
+        const component = await Component.getOpenShiftCmdData(context,
+            'Select a Project',
+            'Select an Application',
+            'Select a Component');
+        if (!component) return null;
+        const linkComponent = await Component.getLinkData(component);
+        const getLinkComponent = linkComponent['status'].linkedComponents;
+        if (!getLinkComponent) throw Error('No linked Components found');
+        Object.keys(getLinkComponent).forEach(async key => {
+            linkCompName.push(key);
+        });
+        const compName = await window.showQuickPick(linkCompName, {placeHolder: "Select a Component to unlink"});
+        if (!compName) return null;
+        return Progress.execFunctionWithProgress(`Unlinking Component`,
+            () => Component.odo.execute(Command.unlinkComponents(component.getParent().getParent().getName(), component.getParent().getName(), component.getName(), compName), component.contextPath.fsPath)
+                .then(() => `Component ${compName} has been successfully unlinked from the component ${component.getName()}`)
```

Comment:
Wrap compName and comoponent.getName() in ```'```. All ```component``` words should start with capital character ```Component``` in this line and next line.